sympy/integrals/integrals.py

```diff
@@ -1026,16 +1026,19 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
 
                 # fall back to heurisch
                 if heurisch is not False:
-                    try:
-                        if conds == 'piecewise':
-                            h = heurisch_wrapper(g, x, hints=[])
-                        else:
-                            h = heurisch_(g, x, hints=[])
-                    except PolynomialError:
-                        # XXX: this exception means there is a bug in the
-                        # implementation of heuristic Risch integration
-                        # algorithm.
-                        h = None
+                    if heurisch is True or not (g.has(Abs) or g.has(sign)):
```

Comment:
It would be better to have this condition in heurisch itself.

Reply:
Problem is that you cannot force then. At least not without introducing a new flag. But I see the point.

Reply:
heurisch should be responsible for knowing what sorts of things it can and cannot handle. That is how every other integrator works (except for the polynomial and rational function ones it looks like, but even those could be made to work this way). There could be some internal heurisch function that bypasses this if someone *really* wants to see if something works anyway.